### PR TITLE
40 proxy docker

### PR DESCRIPTION
**Pull request by vasiliy-zaznobin**
Introduce Dockerfile to create a proxy docker image
Introduce .buildkite/* to support Buildkite
Introduce proxy/docker-compose-test.yml and
          .buildkite/steps/deploy-test.sh
          to run tests in the file proxy/deploy-test.sh
          
Developers can check your changes locally by next steps:
 1. do one-more commits into your local repository
 2. run .buildkite/steps/build-image.sh
 3. run .buildkite/steps/deploy-test.sh
If "Deploy test success" appeared
 then they can push into the remote repository

### CODE REVIEW

File: .buildkite/steps/deploy-test.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+}
+trap cleanup_docker EXIT
+sleep 10
+
+PROXY_URL=http://127.0.0.1:9090/solana
+
+echo "Wait proxy..." && wait-for-proxy "$PROXY_URL"
+echo "Run tests..."
+docker run --rm --network proxy_net -ti \
+     -e SOLANA_URL=http://solana:8899 \
```
Comment (s-medvedev) [inline .buildkite/steps/deploy-test.sh]: SOLANA_URL variable is not required for this tests.

File: Dockerfile [hunk 2]
```diff
@@ -1,28 +1,46 @@
...
+
+RUN python3 -m venv venv
+RUN pip3 install --upgrade pip
+RUN /bin/bash -c "source venv/bin/activate"
+RUN ls .
+RUN pip install -r requirements.txt
+
+ENV PATH /venv/bin:/cli/bin/:/spl/bin/:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+ENV SOLANA_URL="http://localhost:8899"
+#ENV SOLANA_URL="http://37.204.19.33:8899"
```
Comment (s-medvedev) [inline Dockerfile:43]: Please, remove this line.

File: Dockerfile [hunk 3]
```diff
@@ -1,28 +1,46 @@
...
+COPY --from=spl /opt/solana/bin/solana-deploy /cli/bin/
+
+COPY --from=spl /opt/evm_loader.so /spl/bin/
+COPY --from=spl /opt/neon-cli /spl/bin/
+COPY --from=spl /opt/neon-cli /spl/bin/emulator
+
+RUN python3 -m venv venv
+RUN pip3 install --upgrade pip
+RUN /bin/bash -c "source venv/bin/activate"
+RUN ls .
```
Comment (vodolaz) [inline Dockerfile:38]: i think this line can be deleted

File: Dockerfile [hunk 4]
```diff
@@ -1,28 +1,46 @@
...
+RUN DEBIAN_FRONTEND=noninteractive apt -y install \
+                                          software-properties-common \
+                                          openssl \
+                                          ca-certificates \
+                                          curl \
+                                          lsof \
+                                          iputils-ping
+RUN add-apt-repository universe
+RUN apt -y install python3-pip python3-venv
+RUN rm -rf /var/lib/apt/lists/*
```
Comment (vodolaz) [inline Dockerfile:20]: need concatenate all work with packages into one RUN command, it decrace layers count

File: Dockerfile [hunk 5]
```diff
@@ -1,28 +1,46 @@
...
+COPY . /opt
+WORKDIR /opt
+
+RUN apt -y update
+RUN DEBIAN_FRONTEND=noninteractive apt -y install \
+                                          software-properties-common \
+                                          openssl \
+                                          ca-certificates \
+                                          curl \
+                                          lsof \
```
Comment (vodolaz) [inline Dockerfile:16]: remove this packages if it don't need

File: Dockerfile (hunk 4 above)
Comment (s-medvedev) [reply Dockerfile:20]: I sugest to make optimisations in another issue. There are some points which need to fix in Dockerfile.

**Review by vasiliy-zaznobin — COMMENTED** [review]
OK

**Review by s-medvedev — APPROVED** [review]

**Pull request merged**